deployment/terraform/anonymize_db_job.tf

```diff
@@ -0,0 +1,155 @@
+locals {
+ database_anonymizer_image = "${module.ecr_repository_database_anonymizer.repository_url}:${var.image_tag}"
+}
+
+variable "anonymized_db_s3_storage" {
+  type        = string
+  description = "Storage bucket for anonymized database dump"
+  default     = "oshub-dumps-anonymized"
+}
+
+resource "aws_cloudwatch_log_group" "database_anonymizer" {
+  name              = "log${local.short}DatabaseAnonymizer"
+  retention_in_days = 30
+}
+
+data "aws_iam_policy_document" "database_anonymizer_assume_role" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "database_anonymizer_worker" {
+  statement {
+    actions = [
+      "dynamodb:GetItem",
+      "dynamodb:PutItem",
+      "dynamodb:Scan",
+      "logs:CreateLogStream",
+      "logs:CreateLogStream",
+      "logs:DescribeLogGroups",
+      "logs:DescribeLogStreams",
+      "logs:PutLogEvents",
+      "logs:PutLogEvents"
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_role" "database_anonymizer" {
+  name_prefix        = join("-", [local.short, "DatabaseAnonymizer"])
+  assume_role_policy = data.aws_iam_policy_document.database_anonymizer_assume_role.json
+}
+
+resource "aws_iam_policy" "database_anonymizer" {
+  name_prefix = join("-", [local.short, "DatabaseAnonymizer"])
+  path        = "/"
+  policy      = data.aws_iam_policy_document.database_anonymizer_worker.json
+}
+
+resource "aws_iam_role_policy_attachment" "database_anonymizer" {
+  role       = aws_iam_role.database_anonymizer.name
+  policy_arn = aws_iam_policy.database_anonymizer.arn
+}
+
+# AWS ECS Cluster Terraform Module
+# https://registry.terraform.io/modules/cn-terraform/ecs-cluster/aws/latest
+module "database_anonymizer_cluster" {
+  source  = "git::git@github.com:cn-terraform/terraform-aws-ecs-cluster.git?ref=1.0.11"
+  name    = join("-", [local.short, "DatabaseAnonymizer"])
+}
+
+# AWS ECS Fargate Task Definition Terraform Module
+# https://registry.terraform.io/modules/cn-terraform/ecs-fargate-task-definition/aws/latest
+module "database_anonymizer_task_definition" {
+  source                 = "git@github.com:cn-terraform/terraform-aws-ecs-fargate-task-definition.git?ref=1.0.36"
+  name_prefix            = "database-anonymizer"
+  container_image        = local.database_anonymizer_image
+  container_name         = "database-anonymizer"
+  #ephemeral_storage_size = 25
+
+  environment = [
+    {
+      name: "POSTGRES_USER",
+      value: "anondb"
+    },
+    {
+      name: "POSTGRES_PASSWORD",
+      value: "anondb"
+    },
+    {
+      name: "POSTGRES_DB",
+      value: "anon"
+    },
```

Comment:
Are these values real?

Reply:
This temporary database is recreated each time the scheduled task runs.